config/initializers/doorkeeper.rb

```diff
@@ -347,11 +347,34 @@
   grant_flows %w[password]
 
   resource_owner_from_credentials do
-    user = User.find_for_database_authentication(email: params[:email])
-    if user && user.valid_password?(params[:password])
-      user
+    login = params[:email]
+    password = params[:password]
+    user = nil
+
+    # LDAP IS ACTIVE
+    if ENV['DEVISE_LDAP_LOGGER'].present?
+      # Check LDAP credentials
+      if User.check_ldap_credentials(login, password) # ldap valide
+        user = User.find_for_authentication(login: login)
+        #  Check if user exists in database
+        if user.present?
+        else
```

Comment:
If the `if` is empty, you should change to:
```suggestion
        if user.blank?
```